### ISSUE TITLE
Web Dynpro ABAP: Context node attribute referencing field removed from dictionary structure

### ISSUE THREAD

**Issue by kjetil-kilhavn** [larshp/abapGit#3012]
I removed and added fields in a dictionary structure used as reference for a context node in a web dynpro component, and I updated the web dynpro component context node to reflect those changes.
When importing the component in another system the context node attributes were not updated correctly, and therefore the web dynpro component could not be activated after import. 

I checked the XML of the web dynpro component in our git repository, and the serializing is correct, so this is a problem when importing. I get a syntax error for the reference to the (now) non-existing element of the dictionary structure.

**Comment by larshp** [comment]
can it be reproduced in a public repo?

**Comment by kjetil-kilhavn** [comment]
Probably, but I can't make this repository public. 
By the way; I pushed from our sandbox and pulled to our development system, so it's the same system line and same SAP release.

**Comment by kjetil-kilhavn** [comment]
I guess I can try to create local objects and reproduce the problem. Give me a few days if you have the patience :-)

**Comment by larshp** [comment]
make some dummy objects to try reproducing the behavior, will also help to narrow down the example

**Comment by kjetil-kilhavn** [comment]
Yes, that's what I was planning. A dictionary structure with a few fields and a web dynpro component with one view.
The added fields are added to the context node, so the 'only' problem is that the removed attributes are not deleted from the context node. In the same change I also had package restructuring and a lot of other stuff, but hopefully this bug is isolated from all of that.

**Comment by mbtools** [comment]
Same cause as I mentioned in #3013: The delta logic does not take updates into consideration (just inserts).
